// papacarousel/esm2015/lib/carouselNavigation.js
import { concreteSubject } from './helpers/concreteSubject';
import { listener } from './helpers/listener';
export class CarouselNagivation {
    constructor() {
        this.prevSvg = `
        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
                width="24px" height="24px" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
            <path d="M15.41,16.59L10.83,12l4.58-4.59L14,6l-6,6l6,6L15.41,16.59z"/>
            <path fill="none" d="M0,0h24v24H0V0z"/>
        </svg>
    `;
        this.nextSvg = `
        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
                width="24px" height="24px" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
            <path d="M8.59,16.59L13.17,12L8.59,7.41L10,6l6,6l-6,6L8.59,16.59z"/>
            <path fill="none" d="M0,0h24v24H0V0z"/>
        </svg>
    `;
        this.onNavClick = (e) => {
            concreteSubject.notifyAll(this.getDivRef(e.target).className === 'prev-btn' ? -1 : 1);
        };
    }
    initButtons(parentEl) {
        [this.createNavEl(parentEl, this.prevSvg, 'prev-btn'), this.createNavEl(parentEl, this.nextSvg, 'next-btn')]
            .map(b => listener.addListener('click', b, this.onNavClick));
    }
    createNavEl(parentEl, svg, className) {
        const divEl = document.createElement('div');
        divEl.classList.add(className);
        divEl.innerHTML = svg;
        parentEl.appendChild(divEl);
        return divEl;
    }
    getDivRef(el) {
        if (el instanceof HTMLDivElement)
            return el;
        return this.getDivRef(el.parentElement);
    }
}
export const carouselNavigation = new CarouselNagivation;
