// papacarousel/esm2015/lib/papaCarouselBuilder.js
import { PapaCarousel } from './papaCarousel';
export class PapaCarouselBuilder {
    constructor() {
        this.progressBar = false;
        this.buttons = false;
        this.type = 'fading';
        this.timePerSlide = 5000;
        this.transitionFn = 'ease-in';
        this.timingFn = 250;
        this.parent = '.papa-container';
        this.elements = '.papa-item';
    }
    setType(val) {
        this.type = val;
        return this;
    }
    getType() {
        return this.type;
    }
    setParent(val) {
        this.parent = val;
        return this;
    }
    getParent() {
        return this.parent;
    }
    setButtons(val) {
        this.buttons = val;
        return this;
    }
    getButtons() {
        return this.buttons;
    }
    setTimePerSlide(val) {
        this.timePerSlide = val;
        return this;
    }
    setTransitionFn(val) {
        this.transitionFn = val;
        return this;
    }
    getTransitionFn() {
        return this.transitionFn;
    }
    getTimePerSlide() {
        return this.timePerSlide;
    }
    setTimingFn(val) {
        this.timingFn = val;
        return this;
    }
    getTimingFn() {
        return this.timingFn;
    }
    setProgressBar(val) {
        this.progressBar = val;
        return this;
    }
    getProgressBar() {
        return this.progressBar;
    }
    setElements(val) {
        this.elements = val;
        return this;
    }
    getElements() {
        return this.elements;
    }
    getPapa() {
        return new PapaCarousel(this);
    }
}
