// papacarousel/lib/lib/carouselNavigation.js
import { concreteSubject } from './helpers/concreteSubject';
import { listener } from './helpers/listener';
var CarouselNagivation = /** @class */ (function () {
    function CarouselNagivation() {
        var _this = this;
        this.prevSvg = "\n        <svg version=\"1.1\" id=\"Layer_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\"\n                width=\"24px\" height=\"24px\" viewBox=\"0 0 24 24\" enable-background=\"new 0 0 24 24\" xml:space=\"preserve\">\n            <path d=\"M15.41,16.59L10.83,12l4.58-4.59L14,6l-6,6l6,6L15.41,16.59z\"/>\n            <path fill=\"none\" d=\"M0,0h24v24H0V0z\"/>\n        </svg>\n    ";
        this.nextSvg = "\n        <svg version=\"1.1\" id=\"Layer_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\"\n                width=\"24px\" height=\"24px\" viewBox=\"0 0 24 24\" enable-background=\"new 0 0 24 24\" xml:space=\"preserve\">\n            <path d=\"M8.59,16.59L13.17,12L8.59,7.41L10,6l6,6l-6,6L8.59,16.59z\"/>\n            <path fill=\"none\" d=\"M0,0h24v24H0V0z\"/>\n        </svg>\n    ";
        this.onNavClick = function (e) {
            concreteSubject.notifyAll(_this.getDivRef(e.target).className === 'prev-btn' ? -1 : 1);
        };
    }
    CarouselNagivation.prototype.initButtons = function (parentEl) {
        var _this = this;
        [this.createNavEl(parentEl, this.prevSvg, 'prev-btn'), this.createNavEl(parentEl, this.nextSvg, 'next-btn')]
            .map(function (b) { return listener.addListener('click', b, _this.onNavClick); });
    };
    CarouselNagivation.prototype.createNavEl = function (parentEl, svg, className) {
        var divEl = document.createElement('div');
        divEl.classList.add(className);
        divEl.innerHTML = svg;
        parentEl.appendChild(divEl);
        return divEl;
    };
    CarouselNagivation.prototype.getDivRef = function (el) {
        if (el instanceof HTMLDivElement)
            return el;
        return this.getDivRef(el.parentElement);
    };
    return CarouselNagivation;
}());
export { CarouselNagivation };
export var carouselNavigation = new CarouselNagivation;
