// papacarousel/lib/lib/papaCarouselBuilder.js
import { PapaCarousel } from './papaCarousel';
var PapaCarouselBuilder = /** @class */ (function () {
    function PapaCarouselBuilder() {
        this.progressBar = false;
        this.buttons = false;
        this.type = 'fading';
        this.timePerSlide = 5000;
        this.transitionFn = 'ease-in';
        this.timingFn = 250;
        this.parent = '.papa-container';
        this.elements = '.papa-item';
    }
    PapaCarouselBuilder.prototype.setType = function (val) {
        this.type = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getType = function () {
        return this.type;
    };
    PapaCarouselBuilder.prototype.setParent = function (val) {
        this.parent = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getParent = function () {
        return this.parent;
    };
    PapaCarouselBuilder.prototype.setButtons = function (val) {
        this.buttons = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getButtons = function () {
        return this.buttons;
    };
    PapaCarouselBuilder.prototype.setTimePerSlide = function (val) {
        this.timePerSlide = val;
        return this;
    };
    PapaCarouselBuilder.prototype.setTransitionFn = function (val) {
        this.transitionFn = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getTransitionFn = function () {
        return this.transitionFn;
    };
    PapaCarouselBuilder.prototype.getTimePerSlide = function () {
        return this.timePerSlide;
    };
    PapaCarouselBuilder.prototype.setTimingFn = function (val) {
        this.timingFn = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getTimingFn = function () {
        return this.timingFn;
    };
    PapaCarouselBuilder.prototype.setProgressBar = function (val) {
        this.progressBar = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getProgressBar = function () {
        return this.progressBar;
    };
    PapaCarouselBuilder.prototype.setElements = function (val) {
        this.elements = val;
        return this;
    };
    PapaCarouselBuilder.prototype.getElements = function () {
        return this.elements;
    };
    PapaCarouselBuilder.prototype.getPapa = function () {
        return new PapaCarousel(this);
    };
    return PapaCarouselBuilder;
}());
export { PapaCarouselBuilder };
