// papacarousel/lib/lib/papaFascade.js
import { listener } from './helpers/listener';
import { PapaCarouselBuilder } from './papaCarouselBuilder';
var PapaFascade = /** @class */ (function () {
    function PapaFascade() {
        this.papaBuilder = new PapaCarouselBuilder();
    }
    PapaFascade.prototype.destroyListeners = function () {
        listener.destroyListeners();
    };
    PapaFascade.prototype.setParent = function (val) {
        return this.papaBuilder.setParent(val);
    };
    PapaFascade.prototype.setElements = function (val) {
        return this.papaBuilder.setElements(val);
    };
    PapaFascade.prototype.setType = function (val) {
        return this.papaBuilder.setType(val);
    };
    PapaFascade.prototype.setButtons = function (val) {
        return this.papaBuilder.setButtons(val);
    };
    PapaFascade.prototype.setProgressBar = function (val) {
        return this.papaBuilder.setProgressBar(val);
    };
    PapaFascade.prototype.setTimePerSlide = function (val) {
        return this.papaBuilder.setTimePerSlide(val);
    };
    PapaFascade.prototype.setTimingFn = function (val) {
        return this.papaBuilder.setTimingFn(val);
    };
    PapaFascade.prototype.setTransitionFn = function (val) {
        return this.papaBuilder.setTransitionFn(val);
    };
    PapaFascade.prototype.getPapa = function () {
        return this.papaBuilder.getPapa();
    };
    return PapaFascade;
}());
export { PapaFascade };
