// papacarousel/lib/lib/progressBar.js
import { concreteSubject } from './helpers/concreteSubject';
var ProgressBar = /** @class */ (function () {
    function ProgressBar() {
    }
    ProgressBar.prototype.initBar = function (parentEl, timeAnimationFn) {
        concreteSubject.attach(this.resetAnimation.bind(this));
        this.progressBarEl = this.createNavEl(parentEl, ['progress-bar', 'animate'], timeAnimationFn);
    };
    ProgressBar.prototype.createNavEl = function (parentEl, className, timeAnimationFn) {
        var _a;
        var divEl = document.createElement('div');
        (_a = divEl.classList).add.apply(_a, className);
        divEl.style.animationDuration = timeAnimationFn + 'ms';
        parentEl.appendChild(divEl);
        return divEl;
    };
    ProgressBar.prototype.resetAnimation = function () {
        this.progressBarEl.classList.remove('animate');
        void this.progressBarEl.offsetWidth;
        this.progressBarEl.classList.add('animate');
    };
    return ProgressBar;
}());
export { ProgressBar };
export var progressBar = new ProgressBar;
